pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py

```diff
@@ -337,3 +339,147 @@ def _gen_modular_errata():
                 'packages': []
             }]
         }
+
+
+class ModularErrataCopyConservativeTestCase(unittest.TestCase):
+    """Test ``recursive`` and ``recursive_conservative`` flags during copy.
+
+    This test targets the following issues:
+
+    * `Pulp #4518 <https://pulp.plan.io/issues/4518>`_
+    * `Pulp #4548 <https://pulp.plan.io/issues/4548>`_
+
+    Recursive copy of ``RHEA-2012:0059`` should copy:
+
+    * 2 modules: ``duck`` and ``kangaroo``.
+    * 2 RPMS: ``kangaroo-0.3-1.noarch.rpm``, and ``duck-0.7-1.noarch.rpm``.
+
+    Exercise the use of ``recursive`` and ``recursive_conservative``.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.19'):
+            raise unittest.SkipTest('This test requires Pulp 2.19 or newer.')
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_recursive_noconservative_nodependency(self):
+        """Recursive, non-conservative, and no old dependency."""
+        repo = self.copy_modular_errata(True, False)
+        self.make_assertions_nodependency(repo)
+
+    def test_recursive_conservative_nodepdendency(self):
+        """Recursive, conservative, and no old dependency."""
+        repo = self.copy_modular_errata(True, True)
+        self.make_assertions_nodependency(repo)
+
+    def test_norecursive_conservative_nodepdendency(self):
+        """Non-Recursive, conservative, and no old dependency."""
+        repo = self.copy_modular_errata(False, True)
+        self.make_assertions_nodependency(repo)
+
+    def test_recursive_noconservative_dependency(self):
+        """Recursive, non-conservative, and older version of RPM present."""
+        repo = self.copy_modular_errata(True, False, True)
+        self.make_assertions_dependency(repo)
+
+    def test_norecursive_conservative_dependency(self):
+        """Non-recursive, conservative, and older version of RPM present."""
+        repo = self.copy_modular_errata(False, True, True)
+        self.make_assertions_dependency(repo)
+
+    def make_assertions_dependency(self, repo):
+        """Make assertions over a repo with an older version of RPM present."""
+        versions = sorted([
+            unit['metadata']['version']
+            for unit in search_units(self.cfg, repo, {'type_ids': ['rpm']})
+            if unit['metadata']['name'] == 'duck'
+        ])
+
+        # 2 due to the older version already present on the repository.
+        self.assertEqual(len(versions), 2, versions)
+
+        self.assertEqual(
+            repo['content_unit_counts']['erratum'],
+            1,
```

Comment:
I did think about this one to be a constant. It could be added to MODULE_FIXTURES_ERRATA, but it was misleading IMO. Since there are 2 erratas on updatedinfo.xml. 

That is why I added this line to the constants.

```.rst
:data:`MODULE_FIXTURES_ERRATA['total_available_units']` = ``modules + rpm_count + erratum``
```

Reply:
@kersommoura +1